Fix/vpn stats

ersonp opened the pull request:
Did you run `make format && make check`?
yes

Fixes #946 

 Changes:	
- Fixed bandwidth 	

How to test this PR:
1. Start a vpn-server
2. Start a vpn-client and connect to the VPN server
3. Compare the stats in the UI with any networking tool like `nload` on `wtun4` interface

alexadhy committed:
latency fix attempt

alexadhy committed:
Merge remote-tracking branch 'upstream/develop' into vpn-stats

alexadhy committed:
Merge remote-tracking branch 'upstream/develop' into vpn-stats

alexadhy committed:
vpn-stats latency and throughput

alexadhy committed:
unix milli

alexadhy committed:
route_group update

alexadhy committed:
fixes tests

alexadhy committed:
fixes tests

alexadhy committed:
fixes conflict

alexadhy committed:
trying fixed interface

alexadhy committed:
add other packets as well to the overall network bandwidth meter

alexadhy committed:
Merge remote-tracking branch 'upstream/develop' into vpn-stats

alexadhy committed:
Merge remote-tracking branch 'upstream/develop' into vpn-stats

alexadhy committed:
Merge remote-tracking branch 'upstream/develop' into vpn-stats

alexadhy committed:
route group latency unix nano

ersonp committed:
Merge branch 'develop' into fix/vpn-stats

ersonp committed:
Fix AddBandwidthReceived

This commit fixes the usage of AddBandwidthReceived since the only time the code would reach there was if it was a handshake packet since all other cases returned a hander.

Now the AddBandwidthReceived method is moved back to handleDataPacket.

ersonp committed:
Fix linting

ersonp commented:
https://user-images.githubusercontent.com/38801159/167630074-d47def90-4ac2-4bf4-9705-804f3e9eeef8.mp4

The upload and download speeds are now almost the same as shown in the tool `nload` for the tun interface. We send the data in API in bytes which I suppose is converted to `bits` for the upload and download speed in the UI. We need to specify it a bit more clearly that the speed is in terms of `bits` as compared to the total that is in terms of `bytes`.
```
[
   {
      "is_alive":true,
      "latency":196,
      "upload_speed":691,
      "download_speed":561,
      "bandwidth_sent":624761,
      "bandwidth_received":665736,
      "error":"",
      "connection_duration":89
   }
]
```
The latency is still a bit inaccurate (working on it so see if anything more can be done) and shoots up whenever there is a big download (tested this with meter.net) and the ping spikes there too while a download is going on so the behaviour seems normal.
![image](https://user-images.githubusercontent.com/38801159/167633321-63499107-1f25-4b1b-a3b0-0b5049e7d095.png)

The ping to the server the `vpn-server` constantly gives a ping of `240 ms`
![image](https://user-images.githubusercontent.com/38801159/167635126-b2b82601-730a-41aa-a4b6-31a58e5aef83.png)

mrpalide commented:
I test it, and can confirm that **Speed** and **Bandwidth** are OK.

https://user-images.githubusercontent.com/79150699/167779555-ef4d8de2-a353-4281-9a0a-65ba5631417c.mov

But **ping** is a few different from what showing in UI and results from teminal.

![image](https://user-images.githubusercontent.com/79150699/167783964-55e76368-d9fa-4d82-9857-f7226cc9c71b.png)

ersonp committed:
Update latency

This commit saves the latency in the varible latency instead of calculating it with time.Since(sentAt).Milliseconds() which caused the debug log and the SetLatency to be diffrent.

We also convert the sentAt to utc and also calculate latency with UTC. This does not change the output but we do it just to be inline with all other time calculations that are now in UTC.

ersonp committed:
Merge branch 'develop' into fix/vpn-stats

ersonp committed:
Make format

ersonp committed:
Fix appveyor linux test

This commit removes the install command for the libraries gcc libgtk-3-dev libayatana-appindicator3-dev as they are no longer needed after recent changes.
